src/main/scala/org/openhorizon/exchangeapi/route/user/Users.scala

```diff
@@ -214,6 +214,271 @@ trait Users extends JacksonSupport with AuthenticationSupport {
+    parameters = Array(
+      new Parameter(name = "organization", in = ParameterIn.PATH, description = "Organization id.")),
+    responses = Array(
+      new responses.ApiResponse(responseCode = "200", description = "response body",
+        content = Array(
+          
+          new Content(
+            examples = Array(
+              new ExampleObject(
+                value =
+                  """{
+  "users": {
+    "orgid/username": {
+      "password": "string",
+      "admin": false,
+      "email": "string",
+      "lastUpdated": "string",
+      "updatedBy": "string",
+      "apikeys": [
+        {
+          "id": "string",
+          "description": "string",
+          "lastUpdated": "string"
+        }
+      ]
+    }
+  },
+  "lastIndex": 0
+}
+"""
+              )
+            ),
+            mediaType = "application/json",
+            schema = new Schema(implementation = classOf[GetUsersResponse])
+          )
+        )),
+      new responses.ApiResponse(responseCode = "401", description = "invalid credentials"),
+      new responses.ApiResponse(responseCode = "403", description = "access denied"),
+      new responses.ApiResponse(responseCode = "404", description = "not found")))
+  def getUserSelfApikey(@Parameter(hidden = true) identity: Identity2,
+                        @Parameter(hidden = true) organization: String): Route =
+    get {
+      logger.debug(s"GET /orgs/$organization/users/apikey - By ${identity.resource}:${identity.role}")
+      
+      val getUserWithApiKeys: CompiledStreamingExecutable[Query[(MappedProjection[UserRow, (Timestamp, Option[String], String, Boolean, Boolean, Timestamp, Option[UUID], String, Option[String], UUID, String)], Rep[Option[(Rep[String], Rep[UUID], Rep[String])]], Rep[Option[ApiKeys]]), (UserRow, Option[(String, UUID, String)], Option[ApiKeyRow]), Seq], Seq[(UserRow, Option[(String, UUID, String)], Option[ApiKeyRow])], (UserRow, Option[(String, UUID, String)], Option[ApiKeyRow])] =
+        for {
+          users <-
+            Compiled((UsersTQ.filter(user => (user.organization === organization &&
+                                             user.username === identity.username))
+                            .filterIf(identity.isStandardUser)(users => users.user === identity.identifier.get) // Users can see themselves
+                            .filterIf(identity.isOrgAdmin)(users => !users.isHubAdmin && users.organization === identity.organization && !(users.organization === "root" && users.username === "root")) // Organization Admins can see other Org Admins and Users in their Organization. They cannot see Hub Admins, and they cannot see Root.
+                            .filterIf(identity.isHubAdmin)(users => (users.isHubAdmin || users.isOrgAdmin) && !(users.organization === "root" && users.username === "root")) // Hub Admins can see other Hub Admins and Organization Admins system-wide. They cannot see Root, and they cannot see normal users.
```

Comment:
In the first `filter` you match by org and username it should be enough to pinpoint the concrete user. This endpoints should produce the info about the caller and since each user has power to see himself, we should not need any other `filterIf`s.

Just make sure `user.organization === organization === identity.organization && user.username === identity.username`.